src/zarr/v3/sync.py

```diff
@@ -65,22 +69,19 @@ def sync(coro: Coroutine, loop: Optional[asyncio.AbstractEventLoop] = None):
             raise NotImplementedError("Calling sync() from within a running loop")
     except RuntimeError:
         pass
-    result_box: List[Optional[Any]] = [None]
-    event = threading.Event()
-    asyncio.run_coroutine_threadsafe(_runner(event, coro, result_box), loop)
-    while True:
-        # this loops allows thread to get interrupted
-        if event.wait(1):
-            break
-
-    return_result = result_box[0]
+
+    future = asyncio.run_coroutine_threadsafe(_runner(coro), loop)
+
+    done, _ = wait([future])
+    return_result = list(done)[0].result()
```

Comment:
Can you comment on what is happening here?

Some specific questions / ideas:

1. `wait` returns `done` and `not_done`. Do we need to assert anything about `not_done`? I guess we're expecting expecting that `not_done` is always an empty set because we have set `return_when=ALL_COMPLETED`. 
2. Should we consider specifying the `timeout` parameter top `wait`? Would a future that hangs never return here?
3. Can we check that `len(done)` is exactly 1 before indexing out the result?

Reply:
As I understand it, `done, _ = wait([future])` is guaranteed to return `tuple[set with one element, the empty set]`, because we are waiting on a list of futures with 1 element, and `wait` returns when the futures are all done. 

Regarding a timeout, that's a broader question about what error boundaries we want to define for IO operations. At the moment, we basically assume that IO either succeeds or fails after a reasonable amount of time. If we want to include "IO takes forever" as a failure mode, then we would need to start building infrastructure around that higher up in the stack.